zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -716,6 +717,13 @@ private void pRequest2TxnCreate(int type, Request request, Record record, boolea
             ephemeralOwner = EphemeralType.TTL.toEphemeralOwner(ttl);
         } else if (createMode.isEphemeral()) {
             ephemeralOwner = request.sessionId;
+
+            int count = zks.getZKDatabase().getDataTree().getEphemerals(ephemeralOwner).size();
+            int limit = Integer.getInteger("zookeeper.ephemeral.count.limit", 7500);
```

Comment:
That's a good question. I'm not sure if we should enforce a minimum or a maximum.
Though maybe we should allow an override of -1 so that the limit can be ignored.